jetpack-start/wp-cli-api.php

```diff
@@ -0,0 +1,221 @@
+			}
+		}
+
+		$user_id = $user->ID;
+		if ( is_multisite() ) {
+			add_user_to_blog( get_current_blog_id(), $user_id, WPCOM_VIP_MACHINE_USER_ROLE );
+
+			if ( ! is_super_admin( $user_id ) ) {
+				grant_super_admin( $user_id );
+			}
+		} else {
+			if ( ! $user->has_cap( WPCOM_VIP_MACHINE_USER_ROLE ) ) {
+				$user->set_role( WPCOM_VIP_MACHINE_USER_ROLE );
+			}
+		}
+
+		return $user;
+	}
+
+	private function api_request( $url, $args ) {
+		$headers = array_merge( [], $this->get_api_auth_header() );
+		$request = wp_remote_post( $url, [
+			'body'    => $args,
+			'headers' => $headers,
+			'timeout' => 30, // jetpack start can be a bit slow so give it time
+		] );
+
+		if ( is_wp_error( $request ) ) {
+			return $request;
+		}
+
+		$response_code = wp_remote_retrieve_response_code( $request );
+		$body = wp_remote_retrieve_body( $request );
+		if ( 200 !== $response_code ) {
+			return new WP_Error( 'api-request-fail-non-200', sprintf( 'got non-200 response from jetpack start api (code: %d | body: %s)', $response_code, $body ) );
+		}
+
+		$data = json_decode( $body, true );
+		if ( ! $data || ! $data['success'] ) {
+			return new WP_Error( 'api-request-fail-invalid-response', sprintf( 'got invalid response from jetpack start api (body: %s)', $body ) );
+		}
+
+		return $data;
+	}
+
+	private function api_cancel_subscription( $site_id ) {
+		$url = $this->get_api_url( '/cancel-subscription' );
+		$args = [
+			'site_id' => $site_id,
+		];
+
+		return $this->api_request( $url, $args );
+	}
+
+	private function api_fetch_keys( $user ) {
+		$url = $this->get_api_url( '/product-keys' );
+		$args = [
+			'site_url'  => get_site_url(),
+			'site_name' => get_bloginfo( 'name' ),
+			'site_lang' => get_locale(),
+			'user_id'   => $user->ID,
+			'user_role' => WPCOM_VIP_MACHINE_USER_ROLE,
+		];
+
+		return $this->api_request( $url, $args );
+	}
+
+	private function install_keys( $data ) {
+		$runcommand_args = [
+			'exit_error' => false,
+		];
+
+		$akismet_key = $data['akismet_api_key'] ?? '';
+		$akismet_result = WP_CLI::runcommand( sprintf(
+			'jetpack-start keys akismet --akismet_key=%s',
+			escapeshellarg( $akismet_key )
+		), $runcommand_args );
+
+		$vaultpress_key = $data['vaultpress_registration_key'] ?? '';
+		$vaultpress_result = WP_CLI::runcommand( sprintf(
+			'jetpack-start keys vaultpress --vaultpress_key=%s',
+			escapeshellarg( $vaultpress_key )
+		), $runcommand_args );
+
+		$jetpack_id = $data['jetpack_id'] ?? '';
+		$jetpack_secret = $data['jetpack_secret'] ?? '';
+		$jetpack_access_token = $data['jetpack_access_token'] ?? '';
+		$jetpack_result = WP_CLI::runcommand( sprintf(
+			'jetpack-start keys jetpack --jetpack_id=%s --jetpack_secret=%s --jetpack_access_token=%s --user=%s',
+			escapeshellarg( $jetpack_id ),
+			escapeshellarg( $jetpack_secret ),
+			escapeshellarg( $jetpack_access_token ),
+			escapeshellarg( WPCOM_VIP_MACHINE_USER_LOGIN )
+		), $runcommand_args );
```

Comment:
Do we need to be handling errors from these CLI commands, or will the whole process exit if they fail?